### ISSUE TITLE
skipped prompt with interactive flag for some coreutils 

### ISSUE THREAD

**Issue by gforsyth** [scopatz/xonsh#727]
If try to use the `-i` flag with `rm` (to prompt whether or not to delete) the `y/n` prompt is skipped and there's an error.  The same problem is true of `ln` but `ln` may be using `rm` to remove existing files anyway.  

```
> /tmp $ touch foo bar
> /tmp $ ls
bar  foo
> /tmp $ rm -i bar
/usr/bin/rm: remove regular empty file 'bar'? /usr/bin/rm: error closing file
> /tmp $ ln -si foo bar
/usr/bin/ln: replace 'bar'? /usr/bin/ln: error closing file
```

in Bash

```
> /tmp rm -i bar
rm: remove regular empty file 'bar'? 
```

**Comment by scopatz** [comment]
Hi @gforsyth, I am not able to reproduce this either with storing stdout or not.

**Comment by scopatz** [comment]
Here is my rm, by the way:

```console
scopatz@localhost ~/temp $ rm --version
rm (GNU coreutils) 8.21
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Paul Rubin, David MacKenzie, Richard M. Stallman,
and Jim Meyering.
```

**Comment by gforsyth** [comment]
Huh... well, I am a few versions ahead of you 

```console
gil@theo ~ $ rm --version
rm (GNU coreutils) 8.25
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Paul Rubin, David MacKenzie, Richard M. Stallman,
and Jim Meyering.
```

but still that's weird.  Let me see if I can downgrade coreutils and see if it still happens on my machine.

**Comment by gforsyth** [comment]
Ok.  There is something massively weird on my machine, I'm going to close this unless I can make sense of the error and determine that it's actually related to xonsh.

**Comment by scopatz** [comment]
Ok!

**Comment by gforsyth** [comment]
I changed my default shell to `/bin/bash`.  

If I log in and don't `startx`, I can run `xonsh` with or without `--no-rc` and then do a `rm -i tmp/foo` and receive the expected prompt `/usr/bin/rm: remove regular empty file 'tmp/foo'?`.

If I `startx`, then everything goes insane.  I open `uRXVT` to a `bash` prompt, then start `xonsh --no-rc` and I get the following:

```console
gil@theo ~ $ rm -i tmp/foo
/usr/bin/rm: remove regular empty file 'tmp/foo'? n
gil@theo ~ $ rm -i tmp/foo
/usr/bin/rm: remove regular empty file 'tmp/foo'? 
*[1] stopped: rm -i tmp/foo & (20551)
TypeError: unorderable types: NoneType() > int()
gil@theo ~ $ rm -i tmp/foo
/usr/bin/rm: remove regular empty file 'tmp/foo'? n
gil@theo ~ $ rm -i tmp/foo
/usr/bin/rm: remove regular empty file 'tmp/foo'? 
*[2] stopped: rm -i tmp/foo & (22122)
TypeError: unorderable types: NoneType() > int()
gil@theo ~ $ rm -i tmp/foo
/usr/bin/rm: remove regular empty file 'tmp/foo'? 
*[3] stopped: rm -i tmp/foo & (22214)
TypeError: unorderable types: NoneType() > int()
gil@theo ~ $ rm -i tmp/foo
/usr/bin/rm: remove regular empty file 'tmp/foo'? 
*[4] stopped: rm -i tmp/foo & (22390)
TypeError: unorderable types: NoneType() > int()
```

So it mostly doesn't work, but sometimes it does...  If I turn on tracebacks, the `TypeError` exception comes from here:

```console
gil@theo ~ $ rm -i tmp/foo
/usr/bin/rm: remove regular empty file 'tmp/foo'? 
*[2] stopped: rm -i tmp/foo & (13054)
Traceback (most recent call last):
  File "/home/gil/anaconda/lib/python3.5/site-packages/xonsh/base_shell.py", line 150, in default
    self.execer.exec(code, mode='single', glbs=self.ctx)  # no locals
  File "/home/gil/anaconda/lib/python3.5/site-packages/xonsh/execer.py", line 126, in exec
    return exec(code, glbs, locs)
  File "<xonsh-code>", line 1, in <module>
  File "/home/gil/anaconda/lib/python3.5/site-packages/xonsh/built_ins.py", line 655, in subproc_uncaptured
    return run_subproc(cmds, captured=False)
  File "/home/gil/anaconda/lib/python3.5/site-packages/xonsh/built_ins.py", line 638, in run_subproc
    if not prev_is_proxy and hist.last_cmd_rtn > 0 and ENV.get('RAISE_SUBPROC_ERROR'):
TypeError: unorderable types: NoneType() > int()
```

Then the extra bit of crazy.  Note that the `rm -i tmp/foo` jobs are being backgrounded.  

```console
gil@theo ~ $ ls tmp
foo
gil@theo ~ $ rm -i tmp/foo
/usr/bin/rm: remove regular empty file 'tmp/foo'? 
*[1] stopped: rm -i tmp/foo & (7143)
TypeError: unorderable types: NoneType() > int()
gil@theo ~ $ fg
*[1] running: rm -i tmp/foo (7143)
y
gil@theo ~ $ ls tmp
gil@theo ~ $ 
```

So I can foreground it, `STDOUT` from `rm` is skipped and just leaves it waiting for input, which, upon giving it a `y`, then successfully removes `tmp/foo`

If I `chsh` to `xonsh` and then log in (with no `.xonshrc`), I `startx` and then there's no Python exception for the `rm`, and no traceback.

```console
gil@theo ~ $ rm -i tmp/foo
/usr/bin/rm: remove regular empty file 'tmp/foo'? /usr/bin/rm: error closing file
gil@theo ~ $ $XONSH_SHOW_TRACEBACK=True
gil@theo ~ $ rm -i tmp/foo
/usr/bin/rm: remove regular empty file 'tmp/foo'? /usr/bin/rm: error closing file
```

In truth, I don't even know what I'm reporting anymore and I may just have run into a weird sequence of edge cases, but I'm writing it all down here in the event that someone has any idea what is going on.

**Comment by scopatz** [comment]
@gforsyth - I believe that I have a fix for `TypeError: unorderable types: NoneType() > int()` in the `andor` branch.  It would be nice if #672 could be reconciled with master rather than thrown away :)

**Comment by scopatz** [comment]
That is probably asking too much of @adqm and myself :)

**Comment by gforsyth** [comment]
Resolved in #1436 -- for those that find this via the GOOGLE
